.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+        env:
+          CACHELOC: ${{ env.CACHELOC }}
+        if: needs.PREPARE-job.outputs.devmode == 'true'
+        run: |
+          docker buildx build \
+            --progress plain \
+            --cache-from type=local,src=${CACHELOC} \
+            --cache-to type=local,dest=${CACHELOC},mode=max \
+            --provenance=false \
+            --load \
+            --platform ${{ needs.PREPARE-job.outputs.platform }} \
+            --file ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.files }} \
+            --tag ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }} \
+            ${{ github.workspace }}/${{ needs.PREPARE-job.outputs.directory }}
+
+      - name: Save Docker image to tar file
+        run: |
+          docker save -o ${GITHUB_WORKSPACE}/image.tar ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}:${{ needs.PREPARE-job.outputs.date }}
+          echo "Docker image saved to ${GITHUB_WORKSPACE}/image.tar"
+
+      - name: Move image.tar to local persistent storage "$HOME/runner/artifacts"
+        run: |
+          sudo mkdir -p $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}
+          sudo chown -R $(whoami):$(id -gn) $HOME/runner/artifacts/
+          cp ${GITHUB_WORKSPACE}/image.tar $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar
+          echo "Moved image.tar to $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar"
+
+
+  SCAN-AND-REPORT-job:
+    needs: [BUILD-job, PREPARE-job]
+    runs-on: ${{ needs.PREPARE-job.outputs.runner_label }}
+    if: needs.PREPARE-job.outputs.proceed_valid == 'true' && needs.PREPARE-job.outputs.noscan != 'true'
+    steps:
+      - name: Print hostname
+        run: |
+          echo "Hostname: $(hostname)"
+      - name: Copy back persistent storage "$HOME/runner/artifacts" to current directory 
+        run: |
+          if [ -f "$HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar" ]; then
+            echo "File already exists, skipping copy."
+          else
+            cp $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar ${GITHUB_WORKSPACE}
+            echo "Copied image.tar from $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar to ${GITHUB_WORKSPACE}"
+          fi
+      - name: Create Trivy report directory
+        run: mkdir -p ./trivy-reports
+  
+      - name: Scan the Docker image with Trivy
+        uses: aquasecurity/trivy-action@master
+        with:
+          input: './image.tar'        
```

Comment:
I am not very familiar with trivy but does it actually need a tar file to be made?

Reply:
As mentioned above, the tar file is used to speed up data transfer. Otherwise, we need to upload and download the OCI from the remote docker registry.